tests/test_cli_event_source_guards.py

```
@@ -0,0 +1,79 @@
+def validate_incremental_stream_scan(source: str) -> None:
+    start = source.index("    private func readStreamLine(")
+    end = source.index("    private func waitForReadableStream(", start)
+    body = source[start:end]
+
+    required = {
+        "private var streamLineSearchOffset = 0": "persistent scan offset",
+        "streamReadBuffer[searchStart...].firstIndex(of: 0x0A)": "incremental newline search",
+    }
+    for snippet, description in required.items():
+        if snippet not in source:
+            raise AssertionError(f"CLI event reader is missing {description}: {snippet}")
```

Ruff (0.16.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)